esphome/components/iaqcore/iaqcore.cpp

```diff
@@ -0,0 +1,98 @@
+#include "iaqcore.h"
+#include "esphome/core/log.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace iaqcore {
+
+static const char *const TAG = "iaqcore";
+
+enum IAQCoreErrorCode : uint8_t { ERROR_OK = 0, ERROR_RUNIN = 0x10, ERROR_BUSY = 0x01, ERROR_ERROR = 0x80 };
+
+struct SensorData {
+  uint16_t co2;
+  IAQCoreErrorCode status;
+  int32_t resistance;
+  uint16_t tvoc;
+
+  SensorData(const uint8_t *buffer) {
+    this->co2 = (buffer[0] << 8) | buffer[1];
+    this->status = static_cast<IAQCoreErrorCode>(buffer[2]);
+    this->resistance = (buffer[3] << 24) | (buffer[4] << 16) | (buffer[5] << 8) | buffer[6];
```

Comment:
```suggestion
    this->resistance = encode_uint32_t(buffer[3], buffer[4], buffer[5], buffer[6]);
```

Reply:
Woops, this is meant to be this, also with a new include for `helpers.h`
```suggestion
    this->resistance = encode_uint32(buffer[3], buffer[4], buffer[5], buffer[6]);
```